src/core/thread/mle_router.cpp

```diff
@@ -2724,7 +2724,7 @@ otError MleRouter::SendChildIdResponse(Child *aChild)
     SuccessOrExit(error = AppendActiveTimestamp(*message));
     SuccessOrExit(error = AppendPendingTimestamp(*message));
 
-    if (aChild->GetState() != Neighbor::kStateValid)
+    if (aChild->GetRloc16() == 0)
```

Comment:
Just to verify, there is no way we can get here on a device/router having Rloc16 of zero.